### ISSUE TITLE
:bomb: `Bug`: There is an error when destroy the devstream

### ISSUE THREAD

**Issue by jxs1211** [devstream-io/devstream#779]
### What Happened?

Background:
deploy the devstream according to the quick start
env:
windows WSL ubuntu 18.04

There is an error when I try to destroy the devstream by following the quick start step by step, and the previous step is successful.

`[going@dev devstream-quickstart]$ dtm destroy -f quickstart.yaml
2022-06-28 20:05:42 ℹ [INFO]  Destroy started.
2022-06-28 20:05:42 ℹ [INFO]  Got Backend from config: local
2022-06-28 20:05:42 ℹ [INFO]  Using local backend. State file: devstream.state.
2022-06-28 20:05:42 ℹ [INFO]  Tool (githubactions-golang/default) will be deleted.
2022-06-28 20:05:42 ℹ [INFO]  Tool (github-repo-scaffolding-golang/default) will be deleted.
Continue? [y/n]
Enter a value (Default is n): y

2022-06-28 20:05:44 ℹ [INFO]  Start executing the plan.
2022-06-28 20:05:44 ℹ [INFO]  Changes count: 2.
2022-06-28 20:05:44 ℹ [INFO]  -------------------- [  Processing progress: 1/2.  ] --------------------
2022-06-28 20:05:44 ℹ [INFO]  Processing: (githubactions-golang/default) -> Delete ...
2022-06-28 20:05:46 ℹ [INFO]  Deleting GitHub Actions workflow pr-builder.yml ...
2022-06-28 20:05:46 ℹ [INFO]  -------------------- [  Processing aborted.  ] --------------------
2022-06-28 20:05:46 ℹ [INFO]  githubactions-golang/default-Delete -> DELETE https://api.github.com/repos/jxs1211/go-webapp-devstream-demo/contents/.github/workflows/pr-builder.yml: 404 Not Found []
2022-06-28 20:05:46 !! [ERROR]  Destroy failed => some error(s) occurred during plugins destroy process.
[going@dev devstream-quickstart]$ 
`

### How to Reproduce?

exec the destroy cmd after apply action is successful

### Anything else

some detail about my deployment:
`
[going@dev devstream-quickstart]$ tree
.
├── devstream.state
├── quickstart.yaml
└── quickstart.yaml.bak

0 directories, 3 files
[going@dev devstream-quickstart]$ cat quickstart.yaml
---
# core config
varFile: "" # If not empty, use the specified external variables config file
toolFile: "" # If not empty, use the specified external tools config file
state: # state config, backend can be local or s3
  backend: local
  options:
    stateFile: devstream.state

---
# plugins config
tools:
  - name: github-repo-scaffolding-golang
    instanceID: default
    options:
      owner: jxs1211
      repo: go-webapp-devstream-demo
      branch: main
      image_repo: jxs1211/go-webapp-devstream-demo
  - name: githubactions-golang
    instanceID: default
    dependsOn: ["github-repo-scaffolding-golang.default"]
    options:
      owner: jxs1211
      repo: go-webapp-devstream-demo
      language:
        name: go
        version: "1.18"
      branch: main
      build:
        enable: True
      test:
        enable: True
        coverage:
          enable: True
      docker:
        enable: False
[going@dev devstream-quickstart]$
`

### DevStream Version

latest

**Comment by azhsmesos** [comment]
I look at the code, I suspect is in workflow_helper.go file generateGitHubWorkflowFileByName() method have some problem in here, cause I couldn't find the corresponding GitHub plugin cause error when the deleteFile, but I'm not sure if my guess is correct, because I can't debug locally

**Comment by IronCore864** [comment]
@hxcGit please take a look.

@jxs1211 thanks for raising this issue!

**Comment by hxcGit** [comment]
I'm not certain whether this is a reproducible problem. I didn't encounter this error in my environment: `mac-arm64` , `devstream: v0.7.0`. Could you provide more information?

**Comment by azhsmesos** [comment]
So what's the problem

**Comment by hxcGit** [comment]
@azhsmesos  Can you reproduce this problem? It works well in my machine.

**Comment by calvinxu** [comment]
I also met this error when I run "./dtm delete -f quickstart.yaml" on SUSE

2022-07-15 04:13:27 ℹ [INFO]  -------------------- [  Processing aborted.  ] --------------------
2022-07-15 04:13:27 ℹ [INFO]  githubactions-golang/default-Delete -> DELETE https://api.github.com/repos/calvinxu/go-webapp-devstream-demo/contents/.github/workflows/pr-builder.yml: 404 Not Found [].        
2022-07-15 04:13:27 !! [ERROR]  Delete error: some error(s) occurred during plugins delete process.
  -- github.com/devstream-io/devstream/cmd/devstream/delete.go 26
  -- github.com/spf13/cobra@v1.4.0/command.go 860
  -- github.com/spf13/cobra@v1.4.0/command.go 974
  -- github.com/spf13/cobra@v1.4.0/command.go 902
  -- github.com/devstream-io/devstream/cmd/devstream/main.go 100

**Comment by hxcGit** [comment]
@calvinxu  Many thanks to your feedback! This is caused by `calvinxu/go-webapp-devstream-demo/contents/.github/workflows/pr-builder.yml` not being found. Please check whether you delete this repo or this file manually before you use `dtm delete`. If not, please check whether this problem is reproducible and offer more information, such as `os`, `dtm version` and so on.

**Comment by iyear** [comment]
@jxs1211 You are missing the `delete repo/delete repo content` permission for `GITHUB_TOKEN`.

We recently encountered a similar issue #882, and after one-on-one communication on WeChat, we found that the problem lies in incomplete `GITHUB_TOKEN` permissions.

The description of token permissions in our `quickstart` doc only has an external link, and we will improve the `quickstart` doc. Thank you for your hard work and follow up.❤
